modules/c++/re/unittests/test_regex.cpp

```diff
@@ -0,0 +1,313 @@
+/* =========================================================================
+ * This file is part of re-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * re-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <import/re.h>
+#include "TestCase.h"
+#include <map>
+
+TEST_CASE(testCompile)
+{
+    re::Regex rx;
+    // test that an invalid regexp throws an exception
+    try
+    {
+        rx.compile("^(");
+        TEST_FAIL("Should have thrown an exception for this invalid regular expression!");
+    }
+    catch (...)
```

Comment:
Ahh this is lame of TestCase.h.  Can you add a new macro to it `TEST_THROWS` that does a `catch (...)` and then use that here?